src/Sarif/UriHelper.cs

```diff
@@ -69,6 +71,16 @@ public static string MakeValidUri(string path)
                     !path.StartsWith(@"\", StringComparison.Ordinal))
                 {
                     validUri = validUri.Substring(1);
+
+                    // When the UriBuilder constructs an absolute URI, it strips any
+                    // leading "." and ".." segments ("dot-segments", as RFC 3986 calls
+                    // them). Glue them back on so we don't lose the relative path
+                    // information.
+                    string leadingDotSegments = GetLeadingDotSegments(path);
+                    if (!string.IsNullOrEmpty(leadingDotSegments))
+                    {
+                        validUri = leadingDotSegments + validUri;
+                    }
```

Comment:
Sorry that I missed the first code review on this. 

I'm not seeing the explanation anywhere for why this code path (even before this change) for the relative case has a different implementation from what I used in Roslyn. I tried to do it without explicit string manipulation. I assume you found something wrong with that even before the dot-segment issue? Please file a bug against Roslyn for anything that it does not handle correctly vs. your changes.